packages/handler-graphql/src/responses.ts

```diff
@@ -17,13 +17,26 @@ export class ErrorResponse {
         message: string;
         data?: any;
     };
-    constructor(params: ErrorResponseParams) {
+    constructor(params: ErrorResponseParams | (Error & Partial<ErrorResponseParams>)) {
         this.data = null;
-        this.error = {
-            code: params.code || defaultParams.code,
-            message: params.message || defaultParams.message,
-            data: params.data || defaultParams.data
-        };
+
+        if (params instanceof Error) {
```

Comment:
Let's replace this with something like the following:


```ts
this.data = null;

let data = null;
if (params.data || params.stack) {
  data = {};
  if (params.data) {
    data = { ...params.data };
  }

  const debug = process.env.DEBUG === "true";
  if (params.stack && debug) {
    data = { ...data, stack: params.stack };
  }
}

this.error = {
  code: params.code || defaultParams.code,
  message: params.message || defaultParams.message,
  data
};
```

Note that within the backend / GraphQL code, we are almost always not throwing an instance of the native `Error` class, but of [`WebinyError`](https://github.com/webiny/webiny-js/blob/next/packages/error/src/Error.ts#L7), which can contain all three - `message`, `code`, and `data`.

Reply:
Current code would support that well, becasue `WebinyError` extends native `Error` - it even has to, otherwise stack trace would not be preserved. But I'll make the code a little more bit explicit about that.

Reply:
Let's see what you'll end up with and then we'll cover it again if needed. 

P.S. Have in mind that the `data` property of `this.error` should not return either stack trace or data that has been provided by the thrower. It could contain both - some custom data assigned by the thrower and `stack` if the received object contains it and `DEBUG===true`.